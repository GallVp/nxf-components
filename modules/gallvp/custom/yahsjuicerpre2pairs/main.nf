process CUSTOM_YAHSJUICERPRE2PAIRS {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/biopython:1.75':
        'biocontainers/biopython:1.75' }"

    input:
    tuple val(meta), path(tsv)
    path sizes

    output:
    tuple val(meta), path("*.pairs")    , emit: pairs
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    template 'yahsjuicerpre2pairs.py'

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.pairs

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
        biopython: \$(pip list | grep "biopython" | cut -d' ' -f3)
    END_VERSIONS
    """
}
