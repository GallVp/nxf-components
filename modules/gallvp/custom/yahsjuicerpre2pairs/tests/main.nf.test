nextflow_process {

    name "Test Process CUSTOM_YAHSJUICERPRE2PAIRS"
    script "../main.nf"
    process "CUSTOM_YAHSJUICERPRE2PAIRS"

    tag "modules"
    tag "modules_gallvp"
    tag "custom"
    tag "custom/yahsjuicerpre2pairs"
    tag "yahs/juicerpre"

    setup {
        run('YAHS_JUICERPRE') {
            script '../../../yahs/juicerpre/main.nf'
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam' , checkIfExists: true)
                ]

                input[1] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp' , checkIfExists: true)
                input[2] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                """
            }
        }
    }

    test("hongyang - hic") {

        when {
            process {
                """
                input[0] = YAHS_JUICERPRE.out.txt

                def sizes = new File('hongyang.sizes')
                sizes.text = "HYh1_Chr01:6000000-6015000\\t15001\\nHYh2_Chr01:6000000-6015000\\t15001\\n"

                input[1] = sizes.toPath()
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hongyang - hic - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = YAHS_JUICERPRE.out.txt

                def sizes = new File('hongyang.sizes')
                sizes.text = "HYh1_Chr01:6000000-6015000\\t15001\\nHYh2_Chr01:6000000-6015000\\t15001\\n"

                input[1] = sizes.toPath()
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
