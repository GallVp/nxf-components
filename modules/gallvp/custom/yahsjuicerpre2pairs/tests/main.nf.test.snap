{
    "hongyang - hic - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pairs:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,97550bee345676889335aa7c6e2a550b"
                ],
                "pairs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pairs:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,97550bee345676889335aa7c6e2a550b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-11T11:25:00.293571"
    },
    "hongyang - hic": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pairs:md5,fdd4d0b11d1655f8fd99539cd31592c6"
                    ]
                ],
                "1": [
                    "versions.yml:md5,97550bee345676889335aa7c6e2a550b"
                ],
                "pairs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pairs:md5,fdd4d0b11d1655f8fd99539cd31592c6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,97550bee345676889335aa7c6e2a550b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-11T11:08:15.931302"
    }
}